engine/include/cubos/engine/plugins/renderer.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef CUBOS_ENGINE_PLUGINS_RENDERER_HPP
+#define CUBOS_ENGINE_PLUGINS_RENDERER_HPP
+
+#include <cubos/engine/ecs/transform_system.hpp>
+
+#include <cubos/engine/cubos.hpp>
+
+#include <glm/glm.hpp>
+#include <cubos/core/gl/render_device.hpp>
+#include <cubos/core/ecs/entity_manager.hpp>
+
+namespace cubos::engine::plugins
+{
+    /// Plugin to create and use a renderer.
+    ///
+    /// Startup Stages:
+    /// - setRenderer: Sets up the renderer.
+    ///
+    /// Stages:
+    /// - draw: Draws component's to the renderer's frame.
+    /// @param cubos CUBOS. main class
+    void rendererPlugin(Cubos& cubos);
+
+    /// Resource to contain the current active camera
+    struct ActiveCamera
+    {
+        cubos::core::ecs::Entity entity;
```

Comment:
Shouldn't this be a `Camera` component? I believe holding entity references violates the core principle of ECS, so the code that uses this should be changed to work with a direct `Camera` component instead.

Reply:
I think this is fine.
The `ActiveCamera` is a resource which identifies the *entity* which is designated as the active camera, with which we should draw to the screen.
It doesn't store the camera state, it only refers to the entity which stores it.
This makes sense because we can have multiple cameras in a scene and switch between them easily, without removing/adding `Camera` components from them.